hawkbit/src/ddi/common.rs

```diff
@@ -46,3 +51,36 @@ pub enum Finished {
     /// Operation is still in-progress
     None,
 }
+
+pub(crate) async fn send_feedback_internal<T: Serialize>(
+    client: &Client,
+    url: &str,
+    id: &str,
+    execution: Execution,
+    finished: Finished,
+    progress: Option<T>,
+    details: Vec<&str>,
+) -> Result<(), Error> {
+    let mut url: Url = url.parse()?;
+    {
+        match url.path_segments_mut() {
+            Err(_) => {
+                return Err(Error::ParseUrlError(
+                    url::ParseError::SetHostOnCannotBeABaseUrl,
```

Comment:
As this is moving code around not a blocker; but can't this be simplified given the relevant error has has `#[from] url::ParseError` ? i'd expect  e.g. `url.path_segments_mut().map_err(url::ParseError::SetHostOnCannotBeABaseUrl)?` to work

Reply:
Good catch, I added a commit doing that.